### ISSUE TITLE
tf.train.latest_checkpoint fails when paths have "//" 

### ISSUE THREAD

**Issue by pramodkaushik** [tensorflow/tensorflow#16170]
```tf.train.latest_checkpoint``` returns the error 
```
ERROR:tensorflow:Couldn't match files for checkpoint
``` 
when paths in the "checkpoint" file have "//" instead of "/". Usually, good practice of using ```os.path.join``` will help avoid this situation, but I believe TensorFlow should account for '//'s in paths as several developers do not use ```os.path.join```

### To reproduce this error
- Create directory ```/home/user/model```
- Create file ```/home/user/model/checkpoint``` whose contents are 

```
model_checkpoint_path: "/home/user/model//model_1"
all_model_checkpoint_paths: "/home/user/model//model_0"
all_model_checkpoint_paths: "/home/user/model//model_1"
```
- Create empty files ```model/model_1.data-00000-of-00001```, ```model/model_1.index```, ```model/model_1.meta```
- Run ```tf.train.latest_checkpoint('/home/user/model')```. 
Expected output is ```u'/home/user/model//model_1'```, but TF returns an error ```ERROR:tensorflow:Couldn't match files for checkpoint /home/user/model//model_1```

### System information
- OS: Ubuntu 16.04
- TF installed via ```pip install tensorflow-gpu```
- TF version: 1.4.1
- Python version: 2.7 
- CUDA/cuDNN version: 8.0
- GPU model and memory: GeForce GTX 1080, 8GB

### Have I written custom code
Yes

### OS Platform and Distribution
Ubuntu 16.04

### TensorFlow installed from
Installed thru ```pip install tensorflow-gpu```

### TensorFlow version
1.4.1

### Bazel version
N/A

### Exact command to reproduce
N/A

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
Exact command to reproduce

**Comment by khsibr** [comment]
I have noticed the same problem on Ubuntu 16.04.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 119 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
We are closing this issue for now due to lack of activity. Please comment if this is still an issue for you. Thanks!
